SysArchMeter.c

```diff
@@ -0,0 +1,70 @@
+/*
+htop - SysArchMeter.c
+(C) 2021 htop dev team
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+#include "config.h"  // IWYU pragma: keep
+
+#include "SysArchMeter.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+
+static const int SysArchMeter_attributes[] = {HOSTNAME};
+
+static void SysArchMeter_updateValues(Meter* this, char* buffer, size_t size) {
+   static struct utsname uname_info;
+   static int uname_result;
+   static char distro[256];
+   static bool loaded_data = false;
+
+   (void)this;
+
+   if(!loaded_data)
+   {
+      uname_result = uname(&uname_info);
+      FILE* fp = popen("lsb_release -d -s", "r");
+      if(fp != NULL)
+      {
+         if(fgets(distro, 256, fp))
+         {
+            distro[strcspn(distro, "\r\n")] = '\0';
+         }
+         pclose(fp);
+      }
+      loaded_data = true;
+   }
+
+   if(uname_result == 0)
+   {
+      if (strlen(distro) == 0) {
+         snprintf(buffer, size, "%s %s [%s]", uname_info.sysname, uname_info.release,
+                  uname_info.machine);
+      } else {
+         snprintf(buffer, size, "%s %s [%s] / %s", uname_info.sysname, uname_info.release,
+                  uname_info.machine, distro);
+      }
+   }
+   else
+   {
+      snprintf(buffer, size, "Unknown (error)");
```

Comment:
I'd remove the " (error)" as it is quite expected on non-Linux / non-BSD to not be able to identify the system with utsname / lsb_release.

Reply:
Got it.